lib/api/apiUtils/object/storeObject.js

```diff
@@ -26,7 +26,9 @@ function checkHashMatchMD5(stream, hashedStream, dataRetrievalInfo, log, cb) {
             contentMD5,
         });
         log.trace('contentMD5 does not match, deleting data');
```

Comment:
Could you take this change opportunity to turn the above log to an error log, and log the object info with it? I think it's important to have a trace in logs when we delete data because of edge conditions that are not user-triggered. Also the other log could be removed since they kind of duplicate each other.